### PR TITLE
Add cache

### PR DESCRIPTION
**Pull request by hanako-eo**
Add cache on updater and game releases and `cache_lifespan` in config

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,6 +7,7 @@ edition = "2021"
 
 [dependencies]
 actix-web = "4.4"
+cached = { version = "0.48.1", features = ["async"]}
```
Comment (SirLynix) [inline Cargo.toml:10]: please don't set patch version

File: src/main.rs [hunk 2]
```diff
@@ -16,28 +21,45 @@ struct VersionQuery {
...
+    let AppData {
+        cache,
+        config,
+        fetcher,
+    } = app_data.as_ref();
+    let mut cache = cache.lock().unwrap();
+
+    // TODO: remove .cloned
+    let Ok(CachedReleased::Updater(updater_releases)) = cache.try_get_or_set_with("updater_releases", || async {
+        fetcher.get_updater_releases().await.map(CachedReleased::Updater)
```
Comment (SirLynix) [inline src/main.rs:49]: wouldn't it be possible to do it with cached macros?

File: src/main.rs [hunk 3]
```diff
@@ -74,7 +96,11 @@ async fn main() -> Result<(), std::io::Error> {
 
     let bind_address = format!("{}:{}", config.listen_address, config.listen_port);
 
-    let data_config = web::Data::new(AppData { config, fetcher });
+    let data_config = web::Data::new(AppData {
+        cache: Mutex::new(TimedCache::with_lifespan(config.cache_lifespan)), // 5min
```
Comment (SirLynix) [inline src/main.rs:100]: So lifespan is global for every cache entry?

**Review by SirLynix — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 3 above)
Comment (hanako-eo) [reply src/main.rs:100]: yes

File: src/main.rs (hunk 2 above)
Comment (hanako-eo) [reply src/main.rs:49]: it's possible, but the result will be the same problem (for the data) or the creation of 2 different cache structures.

Comment (hanako-eo) [reply src/main.rs:49]: correction, I don't think it's possible to use macros after all because of the asyncrone

File: src/main.rs (hunk 3 above)
Comment (hanako-eo) [reply src/main.rs:100]: I can create 2 caches for the updater and the game if you want

**Comment by SirLynix** [comment]
Thank you for your efforts!

**Pull request merged**